service/storage_proxy.cc

```diff
@@ -3049,7 +3049,9 @@ storage_proxy::create_write_response_handler_helper(schema_ptr s, const dht::tok
 
     auto all = boost::range::join(natural_endpoints, pending_endpoints);
 
-    if (cannot_hint(all, type)) {
+    // If the manager hasn't started yet, no mutation will be performed to another node.
+    // No hint will need to be stored.
+    if (_hints_manager.started() && cannot_hint(all, type)) {
```

Comment:
Shouldn't `_hints_manager.started()` be hidden inside `cannot_hint` implementation?